### ISSUE TITLE
Build error on gcc: 8.2.0

### ISSUE THREAD

**Issue by TenninYan** [qulacs/qulacs#24]
It shows following error.

```
In file included from /Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate_factory.hpp:4,
                 from /Users/tenninyan/Soft/qunasys/qulacs/python/cppsim_wrapper.cpp:30:
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp: In constructor 'ClsParametricRXGate::ClsParametricRXGate(UINT, double)':
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:55: error: unable to find numeric literal operator 'operator""i'
   this->_matrix_element << cos(_angle), sin(_angle) * 1.i, sin(_angle) * 1.i, cos(_angle);
                                                       ^~~
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:55: note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:55: note: or use 'j' instead of 'i' for the GNU built-in suffix
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:74: error: unable to find numeric literal operator 'operator""i'
   this->_matrix_element << cos(_angle), sin(_angle) * 1.i, sin(_angle) * 1.i, cos(_angle);
                                                                          ^~~
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:74: note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:63:74: note: or use 'j' instead of 'i' for the GNU built-in suffix
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp: In constructor 'ClsParametricRZGate::ClsParametricRZGate(UINT, double)':
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:42: error: unable to find numeric literal operator 'operator""i'
   this->_matrix_element << cos(_angle) + 1.i*sin(_angle), 0, 0, cos(_angle) - 1.i * sin(_angle);
                                          ^~~
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:42: note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:42: note: or use 'j' instead of 'i' for the GNU built-in suffix
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:79: error: unable to find numeric literal operator 'operator""i'
   this->_matrix_element << cos(_angle) + 1.i*sin(_angle), 0, 0, cos(_angle) - 1.i * sin(_angle);
                                                                               ^~~
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:79: note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:83:79: note: or use 'j' instead of 'i' for the GNU built-in suffix
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp: In member function 'virtual void ClsParametricPauliRotationGate::set_matrix(ComplexMatrix&) const':
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:120:80: error: unable to find numeric literal operator 'operator""i'
   matrix = cos(_angle)*ComplexMatrix::Identity(matrix.rows(), matrix.cols()) + 1.i * sin(_angle) * matrix;
                                                                                ^~~
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:120:80: note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
/Users/tenninyan/Soft/qunasys/qulacs/include/vqcsim/parametric_gate.hpp:120:80: note: or use 'j' instead of 'i' for the GNU built-in suffix
make[3]: *** [python/CMakeFiles/qulacs.dir/build.make:63: python/CMakeFiles/qulacs.dir/cppsim_wrapper.cpp.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:976: python/CMakeFiles/qulacs.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:141: CMakeFiles/python.dir/rule] Error 2
make: *** [Makefile:190: python] Error 2
Traceback (most recent call last):
  File "setup.py", line 97, in <module>
    'Topic :: Communications :: Email',
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/__init__.py", line 129, in setup
    return distutils.core.setup(**attrs)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 172, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 158, in call_command
    self.run_command(cmdname)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/command/install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 30, in run
    self.build_extension(ext)
  File "setup.py", line 74, in build_extension
    subprocess.check_call(['cmake', '--build', '.', '--target', 'python'] + build_args, cwd=self.build_temp)
  File "/Users/tenninyan/.pyenv/versions/anaconda3-5.1.0/envs/fermi/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--target', 'python', '--config', 'Release', '--', '-j2']' returned non-zero exit status 2.
```

**Comment by corryvrequan** [comment]
According to the following errors, rules about GNU complex literals are changed from gcc-7 to gcc-8.
> note: add 'using namespace std::complex_literals' (from <complex>) to enable the C++14 user-defined literal suffixes
> note: or use 'j' instead of 'i' for the GNU built-in suffix
So, now this lib does not work with gcc-8.
I will fix this issue in the next update.

**Comment by corryvrequan** [comment]
This bug is fixed in #30 .
